crates/bevy_math/src/bounding/bounded2d/primitive_impls.rs

```diff
@@ -0,0 +1,222 @@
+//! Contains [`Bounded2d`](super::Bounded2d) implementations for [geometric primitives](crate::primitives).
+
+use glam::{Mat2, Vec2};
+
+use crate::primitives::{
+    BoxedPolygon, BoxedPolyline2d, Circle, Direction2d, Ellipse, Line2d, Plane2d, Polygon,
+    Polyline2d, Rectangle, RegularPolygon, Segment2d, Triangle2d,
+};
+
+use super::{rotate_vec2, Aabb2d, Bounded2d, BoundingCircle};
+
+impl Bounded2d for Circle {
+    fn aabb_2d(&self, translation: Vec2, _rotation: f32) -> Aabb2d {
+        Aabb2d {
+            min: translation - Vec2::splat(self.radius),
+            max: translation + Vec2::splat(self.radius),
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, self.radius)
+    }
+}
+
+impl Bounded2d for Ellipse {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        //           V = (hh * cos(beta), hh * sin(beta))
+        //      #####*#####
+        //   ###     |     ###
+        //  #     hh |        #
+        // #         *---------* U = (hw * cos(alpha), hw * sin(alpha))
+        //  #            hw   #
+        //   ###           ###
+        //      ###########
+
+        let (alpha, beta) = (rotation, rotation + std::f32::consts::FRAC_PI_2);
+        let (hw, hh) = (self.half_width, self.half_height);
+
+        let (ux, uy) = (hw * alpha.cos(), hw * alpha.sin());
+        let (vx, vy) = (hh * beta.cos(), hh * beta.sin());
+
+        let half_extents = Vec2::new(ux.hypot(vx), uy.hypot(vy));
+
+        Aabb2d {
+            min: translation - half_extents,
+            max: translation + half_extents,
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, self.half_width.max(self.half_height))
+    }
+}
+
+impl Bounded2d for Plane2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        // Add or subtract pi/2 from the rotation to get a direction in the right side of the plane
+        let direction = rotate_vec2(
+            *self.normal,
+            rotation - std::f32::consts::FRAC_PI_2 * self.normal.y.signum(),
+        );
+        let x_parallel = direction == Vec2::X || direction == Vec2::NEG_X;
+        let y_parallel = direction == Vec2::Y || direction == Vec2::NEG_Y;
+
+        // Dividing `f32::MAX` by 2.0 can actually be good so that we can do operations
+        // like growing or shrinking the AABB without breaking things.
+        let half_width = if y_parallel { 0.0 } else { f32::MAX / 2.0 };
+        let half_height = if x_parallel { 0.0 } else { f32::MAX / 2.0 };
+        let half_size = Vec2::new(half_width, half_height);
+
+        Aabb2d {
+            min: translation - half_size,
+            max: translation + half_size,
+        }
+    }
```

Comment:
These are a bit weird, but I think it makes sense? Planes are unbounded by nature, so the extents will just expand infinitely, except if the normal is pointing directly along any of the XYZ axes. We could also just make all extents unbounded, which is what Parry does.

The division by 2 is also a bit odd, but it's something that e.g. Parry does to avoid unexpected results when doing operations on planes. I can remove it if it's too arbitrary/unnecessary though.

I also need to add tests for these, I haven't even tried if this works yet :P